src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -368,6 +369,13 @@ void remove(I key)
     {
         static final BlockCallbacks INSTANCE = new BlockCallbacks();
 
+        private boolean compareProperty(IProperty<?> a, IProperty<?> b)
```

Comment:
`compare` usually means returns `-1`, `0`, `1`, so I think this one should be named `private static boolean arePropertiesEqual`

Also, I think this should use `IProperty.equals`. All the vanilla properties implement that method, so I don't think it's strictly correct to compare just by using `getName` and `getAllowedValues`.

Reply:
Hmm right I guess I can rename. 

As for the equals: The goal was to ensure it has the same states in the same order, not necessarily that it uses the same exact property instance. I see now that Property#equals isn't a reference comparison, but it doesn't seem like it's a valid comparison: it only checks type and name, and does NOT check the list of values, which in this case is extremely important that they are the same list in the same order.

Reply:
Subclasses like `net.minecraft.state.IntegerProperty` and `net.minecraft.state.EnumProperty` check the list of values, it's just not done on the abstract version where it might not matter.

Reply:
.......... I looked 3 times through the code of IntegerProperty specifically, and I didn't see an implementation of equals() there........... Fixing.

Reply:
No worries, that's what code review is for heh